amoro-ams/amoro-ams-server/src/main/java/org/apache/amoro/server/terminal/TerminalManager.java

```diff
@@ -370,7 +380,16 @@ private void applyClientProperties(CatalogMeta catalogMeta) {
         catalogMeta.putToCatalogProperties(
             CatalogMetaProperties.KEY_WAREHOUSE, catalogMeta.getCatalogName());
       } else if (!catalogMeta.getCatalogProperties().containsKey(CatalogProperties.CATALOG_IMPL)) {
-        catalogMeta.putToCatalogProperties("type", catalogType);
+        if (Sets.newHashSet(
```

Comment:
How about setting the `catalog-impl` properties automatically when creating the Glue catalog?

Reply:
Setting the `type` and `catalog-impl` properties at the same time will cause a conflict exception for Iceberg catalog

Reply:
According to the code here, If the `catalog-impl` property exists, we will skip to set the type, so it will be fine.

Reply:
Do you mean we can add the property when build catalog instance?
![image](https://github.com/apache/amoro/assets/54210725/5931eb33-f06e-4b35-8eb9-2499d0c169c9)

Reply:
What I mean is we may better construct the required properties when the catalog is created and persist them in the sysdb rather than put them every time we use them. `AMS_URI` may be different because it may change when AMS reboots.

Reply:
I see. I updated the code, PTAL

Reply:
Thanks for the update.